scrapers/scrape_ow.py

```diff
@@ -1,66 +1,42 @@
 #!/usr/bin/env python3
 
 import re
+import datetime
 from bs4 import BeautifulSoup
 import scrape_common as sc
 
-print('OW')
-d = sc.download('https://www.ow.ch/de/verwaltung/dienstleistungen/?dienst_id=5962', encoding='windows-1252')
-sc.timestamp()
-d = d.replace('&nbsp;', ' ')
-
-# 2020-03-23
-"""
-<p class="object-pages-img"><img src="../../images/5e73948a8f49f.jpg"  alt="Kampagne BAG" style="width:600;height:293;border:0;" /></p><br /><div class="object-pages-description"><p class="icmsPContent icms-wysiwyg-first"><em>Stand 23.03.2020</em></p>
-...
-...
-<h3 class="icmsH3Content"><strong>Anzahl Infizierte</strong></h3>
-
-<p class="icmsPContent">Bisher ist bei 25 Personen im Kanton Obwalden das Coronavirus nachgewiesen worden.</p>
-"""
-
-# 2020-03-27
-"""
-<p class="object-pages-img"><img src="../../images/5e73948a8f49f.jpg"  alt="Kampagne BAG" style="width:600;height:293;border:0;" /></p><br /><div class="object-pages-description"><p class="icmsPContent icms-wysiwyg-first"><em>Stand 27.03.2020</em></p>
-...
-...
-<h3 class="icmsH3Content"><strong><a id="Fallzahl" name="Fallzahl"></a>Fallzahl Kanton Obwalden</strong></h3>
-
-<p class="icmsPContent">Bisher ist bei 37&nbsp;Personen im Kanton Obwalden das Coronavirus nachgewiesen worden. Bereits genesene Personen sind in dieser Zahl ebenfalls enthalten&nbsp;(Stand: 27. M<E4>rz 2020).</p>
-"""
-
-# 2020-04-06
-"""
-<h3 class="icmsH3Content"><strong><a id="Fallzahl" name="Fallzahl"></a>Fallzahl Kanton Obwalden (Stand 6. April 2020, 15.00 Uhr)</strong></h3>
-
-<ul>
-	<li>Positiv getestete Personen: 60</li>
-	<li>In Obwalden hospitalisierte Personen: 1</li>
-	<li>Verstorbene Personen: 0</li>
-</ul>
-"""
-
-
-print('Date and time:',
-      sc.find(r'Stand ([^<]+ Uhr)', d) or sc.find(r'Stand ([^<]+)<', d))
-
+d = sc.download('https://www.ow.ch/de/kanton/publired/publikationen/?action=info&pubid=20318',
+                encoding='windows-1252', silent=True)
 soup = BeautifulSoup(d, 'html.parser')
-i = 0
-for row in soup.find(id='Fallzahl').find_next('table').find_all('tr'):
-    i += 1
+xls_url = soup.find(href=re.compile("\.XLSX$")).get('href')
+assert xls_url, "URL is empty"
+
+for row in soup.find_all('tr'):
     cells = row.find_all('td')
-    assert len(cells) >= 2, "Number of columns changed, not 2+"
-    if i == 1: # Header.
-        assert 'COVID-19' in str(cells[0].find('strong').string)
-        assert 'Anzahl' in str(cells[1].find('strong').string)
-        # assert 'Veränderung' in str(cells[2].find('strong').string)
-        continue
+    if cells[0].string and cells[0].string.startswith('Datum'):
+        file_date = cells[1].string
+
+xls = sc.xlsdownload(xls_url, silent=True)
+rows = sc.parse_xls(xls, header_row=0)
```

Comment:
if you pass header_row=4, you should get nice headers (something like `row['Infizierte Personen']` instead of `row['B']`

Reply:
Thanks for the suggestion. Updated